plume-file-web-upload-jersey/src/main/java/com/coreoz/plume/file/validator/FileUploadValidators.java

```diff
@@ -0,0 +1,62 @@
+package com.coreoz.plume.file.validator;
+
+import com.coreoz.plume.file.data.FileUploadMetadata;
+import com.coreoz.plume.file.utils.FileNameUtils;
+import com.coreoz.plume.jersey.errors.WsError;
+import com.coreoz.plume.jersey.errors.WsException;
+import org.apache.commons.lang3.StringUtils;
+import org.glassfish.jersey.media.multipart.FormDataBodyPart;
+
+import java.util.List;
+import java.util.Objects;
+
+public class FileUploadValidators {
+
+    private FileUploadValidators() {
+        // empty constructor to hide the implicit one
+    }
+
+    public static void verifyRequiredMetadata(FileUploadMetadata fileMetadata) {
+        Objects.requireNonNull(fileMetadata);
+        Objects.requireNonNull(fileMetadata.getMimeType());
+    }
+
+    public static void verifyFileNameLength(FormDataBodyPart formDataBodyPart, long fileMaxSize) {
+        String fileName = formDataBodyPart.getContentDisposition().getFileName();
+        Objects.requireNonNull(fileName);
+        if (fileName.length() > fileMaxSize) {
+            throw new WsException(WsError.REQUEST_INVALID, "Filename too long");
+        }
+    }
+
+    public static void verifyFileSize(FormDataBodyPart formDataBodyPart, long fileMaxSize) {
+        long fileSize = formDataBodyPart.getFormDataContentDisposition().getSize();
+        if (fileSize > fileMaxSize) {
+            throw new WsException(WsError.REQUEST_INVALID, "File too large");
+        }
+    }
+
+    public static void verifyFileExtension(FormDataBodyPart formDataBodyPart, List<String> authorizedExtension) {
+        String fileExtension = FileNameUtils.getExtensionFromFilename(
+            formDataBodyPart.getContentDisposition().getFileName()
+        );
+        if (StringUtils.isEmpty(fileExtension)) {
+            throw new NullPointerException();
```

Comment:
Euhh c'est méchant ça. Tu veux pas juste permettre de traiter les 2 cas :
- fichier. => extension vide
- fichier => extension nulle

Ensuite tu documentes, et comme ça les projets peuvent ajouter ou pas null ou vide

Reply:
pas compris : concrètement ici, si authorizedExtensions permet le .xlsx, mais qu'il n'y a pas d'extension dans le fichier, comment veux-tu que le validateur réagisse ? Moi ca me paraissait OK de dire que si'il n'y a pas d'extensions, alors que je cherche à la vérifier, alors NPE

Reply:
En fait je pensais à l'utiliser cette manière : validateExtensions(file, Set.of(null, "xls"));

Pour dire qu'on peut soit avec un fichier XLS soit ne pas avoir d'extension du tout.
Tu vois ?

Et que surtout si ça valide pas, ça retourne seulement une WsException, pas un nullpointer qui va remonter en erreur interne alors que c'est vraiment une erreur de validation

Reply:
le truc c'est que requireNonNull ca va faire un NPE, c'est pour ca que je me suis permit :p